packages/ckb-indexer/src/transactionCollector.ts

```diff
@@ -0,0 +1,442 @@
+    return { objects, lastCursor };
+  }
+
+  private getTransactionListFromRpc = async (
+    transactionHashList: GetTransactionsResults
+  ) => {
+    const getDetailRequestData = transactionHashList.objects.map(
+      (hashItem: GetTransactionsResult, index: number) => {
+        return {
+          id: index,
+          jsonrpc: "2.0",
+          method: "get_transaction",
+          params: [hashItem.tx_hash],
+        };
+      }
+    );
+    const transactionList: TransactionWithIO[] = await requestBatch(
+      this.CKBRpcUrl,
+      getDetailRequestData
+    ).then((response: GetTransactionRPCResult[]) => {
+      return response.map(
+        (item: GetTransactionRPCResult): TransactionWithIO => {
+          if (!this.filterOptions.skipMissing && !item.result) {
+            throw new Error(
+              `Transaction ${
+                transactionHashList.objects[item.id].tx_hash
+              } is missing!`
+            );
+          }
+          const ioType = transactionHashList.objects[item.id].io_type;
+          const ioIndex = transactionHashList.objects[item.id].io_index;
+          return { ioType, ioIndex, ...item.result };
+        }
+      );
+    });
+    return transactionList;
+  };
+
+  private getCellByOutpoint = async (output: OutPoint) => {
+    const transaction: Transaction = await request(
+      this.CKBRpcUrl,
+      "get_transaction",
+      [output.tx_hash]
+    );
+    return transaction.outputs[parseInt(output.index)];
+  };
+
+  private isCellScriptArgsValidate = (targetCell: Output) => {
+    let lockArgsLen: number | "any" | undefined;
+    let typeArgsLen: number | "any" | undefined;
+    if (this.queries.lock) {
+      lockArgsLen = instanceOfScriptWrapper(this.queries.lock)
+        ? this.queries.lock.argsLen
+        : this.queries.argsLen;
+    }
+    if (this.queries.type && this.queries.type !== "empty") {
+      typeArgsLen = instanceOfScriptWrapper(this.queries.type)
+        ? this.queries.type.argsLen
+        : this.queries.argsLen;
+    }
+    const resultMap: cellFilterResult = {
+      lockArgsLen: true,
+      typeArgsLen: true,
+      isTypeScriptEmpty: true,
+    };
+    if (this.queries.lock && instanceOfScriptWrapper(this.queries.lock)) {
+      resultMap.lockArgsLen = false;
+      if (getHexStringBytes(targetCell.lock.args) !== lockArgsLen) {
+        resultMap.lockArgsLen = true;
+      }
+    }
+    if (this.queries.type && this.queries.type !== "empty") {
+      resultMap.typeArgsLen = false;
+      if (
+        targetCell.type &&
+        getHexStringBytes(targetCell.type.args) !== typeArgsLen
+      ) {
+        resultMap.typeArgsLen = true;
+      }
+    }
+    if (this.queries.type && this.queries.type === "empty") {
+      resultMap.isTypeScriptEmpty = false;
+      if (!targetCell.type) {
+        resultMap.isTypeScriptEmpty = true;
+      }
+    }
+    return (
+      resultMap.lockArgsLen &&
+      resultMap.typeArgsLen &&
+      resultMap.isTypeScriptEmpty
```

Comment:
why a valid script cell's `isTypeScriptEmpty` must be true

Reply:
refactor isCellScriptArgsValidate method, and all changes from last review can be found at https://github.com/sighWang/lumos/pull/2